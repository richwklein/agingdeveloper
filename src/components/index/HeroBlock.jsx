import React, {Fragment} from "react";
import {GatsbyImage} from "gatsby-plugin-image";
import {Box, Card, CardActionArea, CardContent, CardHeader, CardMedia, Grid, Typography} from "@mui/material";
import InternalLink from "../common/InternalLink";
import PropTypes from "prop-types";

const HeroBlock = ({hero: {title, slug, excerpt, image}}) => {
  return (
    <Card
      variant="outlined"
      sx={{
        position: "relative",
        backgroundColor: "primary.main",
        color: "primary.contrastText",
      }}
    >
      <CardActionArea disableRipple component={InternalLink} to={`/article/${slug}`}>
        <CardMedia>
          <GatsbyImage image={image} sx={{zIndex: 1}} />
        </CardMedia>
        <Box
          sx={{
            position: "absolute",
            top: 0,
            left: 0,
            width: "100%",
            height: "100%",
            zIndex: 2,
            backgroundColor: "rgb(55, 71, 79, 0.5)",
            backdropFilter: "blur(4px)",
            overflowY: {xs: "scroll", sm: "hidden"},
          }}
        >
          <Grid container>
            <Grid item md={6} sm={12} xs={12}>
              <Box
                sx={{
                  p: {xs: 1, md: 4},
                  pr: {md: 0},
                }}
              >
                <CardHeader titleTypographyProps={{"variant": "h3", "color": "inherit"}} title={title} />
                <CardContent>
                  <Typography variant="h5" color="inherit" paragraph>
                    {excerpt}
                  </Typography>
                  <Typography variant="subtitle1" color="inherit">
                    <Fragment>{"Continue Reading"}&hellip;</Fragment>
                  </Typography>
                </CardContent>
              </Box>
            </Grid>
          </Grid>
        </Box>
      </CardActionArea>
    </Card>
  );
};

HeroBlock.propTypes = {
  hero: PropTypes.shape({
    title: PropTypes.string.isRequired,
    excerpt: PropTypes.string.isRequired,
    slug: PropTypes.string.isRequired,
    image: PropTypes.any,
  }).isRequired,
};

export default HeroBlock;
