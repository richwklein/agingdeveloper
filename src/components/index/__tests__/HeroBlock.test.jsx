import React from "react";
import {cleanup, render} from "@testing-library/react";
import HeroBlock from "../HeroBlock";

// TODO improve tests
describe("HeroBlock", () => {
  afterEach(cleanup);

  const renderComponent = () => render(
      <HeroBlock hero={{
        "title": "This is an Aritcle Title",
        "excerpt": "This is content from the article.",
        "slug": "2023-12-01-slug",
      }} />,
  );

  test("that it renders correctly.", () => {
    const {container} = renderComponent();
    expect(container.firstChild).toMatchSnapshot();
  });
});
