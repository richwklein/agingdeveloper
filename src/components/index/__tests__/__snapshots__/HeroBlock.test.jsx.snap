// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeroBlock that it renders correctly. 1`] = `
<div
  class="MuiPaper-root MuiPaper-outlined MuiPaper-rounded MuiCard-root css-1gyh4ns-MuiPaper-root-MuiCard-root"
>
  <a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways MuiButtonBase-root MuiCardActionArea-root css-1wtuwsy-MuiTypography-root-MuiLink-root-MuiButtonBase-root-MuiCardActionArea-root"
    data-link="internal"
    href="/article/2023-12-01-slug"
    tabindex="0"
  >
    <div
      class="MuiCardMedia-root css-16cvb1l-MuiCardMedia-root"
    />
    <div
      class="MuiBox-root css-1n5d35j"
    >
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-12 MuiGrid-grid-md-6 css-1fyyp8j-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-1chv412"
          >
            <div
              class="MuiCardHeader-root css-185gdzj-MuiCardHeader-root"
            >
              <div
                class="MuiCardHeader-content css-1qbkelo-MuiCardHeader-content"
              >
                <span
                  class="MuiTypography-root MuiTypography-h3 MuiCardHeader-title css-s3bcgd-MuiTypography-root"
                >
                  This is an Aritcle Title
                </span>
              </div>
            </div>
            <div
              class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
            >
              <p
                class="MuiTypography-root MuiTypography-h5 MuiTypography-paragraph css-187jqjf-MuiTypography-root"
              >
                This is content from the article.
              </p>
              <h6
                class="MuiTypography-root MuiTypography-subtitle1 css-1yqpi80-MuiTypography-root"
              >
                Continue Reading
                â€¦
              </h6>
            </div>
          </div>
        </div>
      </div>
    </div>
    <span
      class="MuiCardActionArea-focusHighlight css-1v2exvi-MuiCardActionArea-focusHighlight"
    />
  </a>
</div>
`;
